netlogo-gui/src/main/prim/etc/_extractrgb.java

```diff
@@ -3,19 +3,38 @@
 package org.nlogo.prim.etc;
 
 import org.nlogo.api.LogoException;
+import org.nlogo.api.AgentException;
+import org.nlogo.api.LogoListBuilder;
 import org.nlogo.core.LogoList;
 import org.nlogo.core.Syntax;
+import org.nlogo.core.SyntaxJ;
 import org.nlogo.nvm.Context;
 import org.nlogo.nvm.Reporter;
 
 public final strictfp class _extractrgb
     extends Reporter {
 
-
   @Override
   public Object report(final org.nlogo.nvm.Context context) throws LogoException {
-
-    return report_1(context, argEvalDoubleValue(context, 0));
+    try{
+      double d = argEvalDoubleValue(context, 0);
+      return report_1(context, d);
+    }
+    catch(LogoException l){
+      org.nlogo.api.Exceptions.ignore( l ) ;
+      LogoList rgb = argEvalList(context, 0);
+      try{
+        org.nlogo.api.Color.validRGBList(rgb,true);
+      }
+      catch(AgentException a){
+        throw new org.nlogo.nvm.RuntimePrimitiveException(context, this, "Color must be a number or a valid RGB(A) List.");
```

Comment:
Move the error message into the internationalization files, so they can be internationalized at some future point.  There are [two](https://github.com/NetLogo/NetLogo/blob/hexy/shared/i18n/Errors_en.txt) [spots](https://github.com/NetLogo/NetLogo/blob/hexy/netlogo-gui/project/autogen/i18n/Errors_en.txt) for this, oddly.  You can see they organize their "access strings" by prim package and name.  [Here is a sample usage](https://github.com/NetLogo/NetLogo/blob/hexy/netlogo-gui/src/main/prim/etc/_sublist.java#L23-L24).

Change the error message to be a little more helpful, something like "...number or a valid RGB(A) list with 3 or 4 number items with values from 0 to 255." to be explicit about what an "RGB(A) list" actually is.